astropy/coordinates/builtin_frames/galactocentric.py

```diff
@@ -126,15 +142,57 @@ class Galactocentric(BaseCoordinateFrame):
              ( 289.77285255,  50.06290457,  8.59216010e+01)]>
 
     """
+    frame_specific_representation_info = {
+        CartesianDifferential: [
+            RepresentationMapping('d_x', 'v_x', u.km/u.s),
+            RepresentationMapping('d_y', 'v_y', u.km/u.s),
+            RepresentationMapping('d_z', 'v_z', u.km/u.s),
+        ],
+    }
+
     default_representation = CartesianRepresentation
+    default_differential = CartesianDifferential
 
-    # TODO: these can all become QuantityFrameAttribute's once #3217 is merged
-    galcen_distance = FrameAttribute(default=8.3*u.kpc)
-    galcen_ra = FrameAttribute(default=Angle(266.4051*u.degree))
-    galcen_dec = FrameAttribute(default=Angle(-28.936175*u.degree))
+    # frame attributes
+    galcen = CoordinateAttribute(default=galcen_default, frame=ICRS)
     z_sun = FrameAttribute(default=27.*u.pc)
     roll = FrameAttribute(default=0.*u.deg)
 
+    def __init__(self, *args, **kwargs):
+
+        # backwards-compatibility
+        if ('galcen_distance' in kwargs or 'galcen_ra' in kwargs or
+                'galcen_dec' in kwargs):
+            warnings.warn("The arguments 'galcen_distance', 'galcen_ra', and "
+                          "'galcen_dec' are deprecated in favor of specifying "
+                          "the full-space position and velocity of the Galactic "
+                          "center by passing a frame in to the 'galcen' attribute",
+                          AstropyDeprecationWarning)
+
+        galcen_kw = dict()
+        galcen_kw['distance'] = kwargs.pop('galcen_distance',
+                                           self.galcen.distance)
+        galcen_kw['ra'] = kwargs.pop('galcen_ra', self.galcen.ra)
+        galcen_kw['dec'] = kwargs.pop('galcen_dec', self.galcen.dec)
+        galcen_kw['pm_ra'] = self.galcen.pm_ra
+        galcen_kw['pm_dec'] = self.galcen.pm_dec
+        galcen_kw['radial_velocity'] = self.galcen.radial_velocity
+        kwargs['galcen'] = ICRS(**galcen_kw)
```

Comment:
Yes, this is just for backwards compatibility (in <=v1.3 the keywords were individual quantities). Though I'm not totally sure on this because if the user just wants to change the default `distance`, they have to set all of the other frame components as well. If you have any ideas...